### DISCUSSION TITLE
[Feature] Memories showcase

### DISCUSSION THREAD

**Discussion by alexIslander** [immich-app/immich#2836] [Feature Request]
### The feature

Just like in Google Photos:
- Memories automatically creates collections of your photos and videos.
Memories are based on factors like location, date, and people.
- Memories can be viewed in the Memories tab of the Google Photos app.
- Machine learning techniques like facial recognition, object recognition, and location detection are used to create memories.
- You can customize which photos and videos are included in memories through settings.
- Specific locations or events can be chosen to include in memories.
- Certain people or objects can be excluded from memories.

I believe immich-ml module could do it! 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jahopki** [comment]
The ability to control which photos are included in memories would be very helpful. Perhaps if we had the ability to "hide" images or even entire days, then those hidden images would be excluded from consideration in presentation as memories...

**Comment by hischampion** [comment]
Love these videos on iCloud and Google. Cue the random Fiver licensed music.

**Comment by vide** [comment]
This Is a good feature request. Any plan to put it in the roadmap? 
Creating automated videos from photos stitching together pictures on the same theme or person over a period of time is basically the only feature I miss from The app that cannot be mentioned. I'm a new Immich user and the wow-factor has been already gigantic for a self-hosted solution. This feature would be the cherry on the cake.

**Comment by petersuba** [comment]
I am trying to wean off my other half from paying for cloud storage of over 1TB of photos/videos and by now this has become her trump card over immich, having shareable short videos of memories that are automatically created based on various factors.

Making it shareable could also make it an effective marketing tool if by default we add an immich logo in the corner...

The features I am seeing in that which may be pulled together for this are:
- Memories as generated exportable/shareable short videos instead of just in-platform slideshows
- A collection of different types of transitions between assets to make it more interesting (crossfade, move over, etc etc)
- Subtle animation effects for stills when showing in memories (i.e. zoom, move left/right etc)
- Text overlays showing relevant metadata or memory grouping rules applied (i.e. together over the years, trip to XYZ etc) and positioning of this text automatically strategically (i.e. avoiding overlaying a face or relevant picture content)
- Adding mood music, preferably not completely random but matching the sentiment of the underlying memory selection (i.e. uplifting music for a trip, sentimental for memory over time etc)

In terms of memories based on factors, could be things like
- on a birthday of someone, a memory for that person, weighing heavily on birthday pictures (queue ML of candles and birthday cakes)
- Pick persons that have lots of photos over years, then pick "good" pictures of them (queue ML smile detection, sharp pictures only, prefer "favorited" etc) together over time
- On this day (that's more or less the current algorithm as far as I understand)
- Trip to X place - maybe trip to X, Y years ago today - assets from a location that is not "typical" in the library, from a period of maybe a week ahead/before this day X years ago (i.e. not just "on this day")

**Comment by Frankster-NL** [comment]
This is exactly the functionality that would change the WAF here fully towards Immich. +1 for me too!

**Comment by MrScripter85** [comment]
+1 from me. I am currently migrating from Google to Immich and those collages as well as the "highlight videos" (the ones that are slowed down, 14 seconds long and have some music) are awesome. This would be the only reason to still keep my images with Google, but I hope that one day, Immich can do the same. I would be even happy to just have a function to create collages based on time/date/location/people.

**Comment by Wired11103** [comment]
Love this idea. Maybe we could create a name for the memories as well. E.g Roger's 7th birthday

**Comment by jagfirerwalker** [comment]
Yes that is the killer feature missing. What would be needed to make that a reality?

**Comment by luisgdev** [comment]
Is there a way to enable this feature, after it was disabled or hidden?

**Reply by kgmd111** [reply]
If I know this is not finished yet( that's why it has feature request flag )

**Reply by luisgdev** [reply]
The solution I found was to create a new Immich job for memory generation. Going to Administration/Jobs/Create new job/Memory generation.

**Reply by dvdstelt** [reply]
I don't see the job appearing however?

**Reply by dvdstelt** [reply]
Doesn't matter, it works anyway ;-)
Too bad no daily notification on my phone. Other than that, nice feature!

**Reply by skatsubo** [reply]
Re: missing memories.
If the memories feature is enabled, but memories not showing up, you may have encountered a bug.

See https://github.com/immich-app/immich/issues/16484#issuecomment-2692718993 for proposed resolution.

At first, you may try only the job part:
> run the memories generation jobs

If that does not help, then do what's described in the linked comment, starting with:
> run the following command to clean the state of the memories generation

and then re-run the job.

**Comment by Guoyuer** [comment]
Generating a short video like apple photos from selected pictures/videos would be a very nice feature!

**Comment by 1661410107114145152525** [comment]
Oh I thought this is already included. This feature is so important for everyone with more than just a few hundreds of photos as otherwise there will be many you will never see again...

**Comment by luna1611** [comment]
I agree.
The ONLY reason that I still keep google photos, while I copied all my photos to immich, is the generation of such memories (collage, video, pseudo-3D picture, and many possibilities considering the current progresses in AI-assisted image control).

**Reply by luna1611** [reply]
In addition, it would be great if we can "request" such images to our own immich server whenever we want.

In google photo, as far as I understand, we CAN make album and some collage whenever we want,
but we CANNOT make more interesting product.
We have nothing but to wait for google photo's random product.

Since immich runs in a personal server, it would be great if we can request such product any time.

Anyway, thanks the development team for their wonderful work!

**Comment by chrneu** [comment]
I think it would also make sense to have an option to exclude screenshots from the memories creation job.

**Reply by ti-guru** [reply]
Adding to the comment above: Also exclude images of documents, power points etc.

**Reply by 1661410107114145152525** [reply]
> Adding to the comment above: Also exclude images of documents, power points etc.

Feels like overkill. Why not just having the possibility to whitelist or blacklist specific tags. As you can already use the smart search to tag or create albums for specific images you don't wan't to include and give specific tags, this should be easier to implement while being flexible enough for all future requests of image typesnpeople might want to exclude.

**Reply by SinTan1729** [reply]
Yes. Excluding tags and poeple should be enough for my use case, but maybe a general way to filter using any criteria might be nice. For now, I'm using a script, but it has limitations since the memories API has limitations.

**Reply by ti-guru** [reply]
> > Adding to the comment above: Also exclude images of documents, power points etc.
> 
> Feels like overkill. Why not just having the possibility to whitelist or blacklist specific tags. As you can already use the smart search to tag or create albums for specific images you don't wan't to include and give specific tags, this should be easier to implement while being flexible enough for all future requests of image typesnpeople might want to exclude.

Agreed. As long as it's easy to exclude unimportant stuff from memories. E.g. don't want to be reminded of a document I photographed X years ago 🤷

**Comment by TimmyChannel** [comment]
The request to add this feature was made two years ago. I think it can be a killer feature for g photos. Are there any plans in this?

**Reply by bo0tzz** [reply]
The age of the feature request isn't particularly meaningful. Fancy memories would be nice, but we currently have no plans to add it.

**Reply by TimmyChannel** [reply]
Thank you for the response, @bo0tzz. I completely agree that the age of the request isn't the key factor - what matters is how this could elevate Immich as a true self-hosted alternative to G Photos. For many of us (myself included), advanced memories aren't just "nice"; they're a core reason we stick with Google despite wanting to migrate. The basic "X years ago" slideshows are a start, but features like automated highlight videos (with transitions, music, and text overlays), collages based on events/people/locations, and customizable exclusions (e.g., hiding screenshots or specific tags) would make Immich unbeatable.
If resources are the holdup, is there any way the community could contribute more directly, like through bounties or PRs? Or perhaps prioritize it in the roadmap if we gather more feedback? I'd love to hear if there's any openness to revisiting this as a potential killer feature.

**Comment by xuxusheng** [comment]
I'd like this feature as well. Right now, all mobile operating systems have this feature built - in. I'm hoping Immich will be able to support it too.

**Comment by Alessio-Olivieri** [comment]
So... is this not implemented yet? My mom would love to be able to see the pictures of when i was young

**Comment by Nordlicht-13** [comment]
Check out this:
https://github.com/ismaildakrory/immich-memories-notify

**Reply by 1661410107114145152525** [reply]
@bo0tzz In your roadmap you spoke about plugin possibilities coming to Immich: Will sth. like that be possible than and do you know if plugins will be reviewed from a security pov? I personally do not need this feature but I am curious how powerful plugins might be. :)

PS: At anyone disliking @bo0tzz reply that it is not planned above: Why doing so instead of appreciating honest answers over empty promises.

**Reply by bo0tzz** [reply]
We're aiming for plugins to eventually be able to make external calls to trigger things like this as well, yes. We don't have the bandwidth to do a security review of every published plugin, but they will all be sandboxed in WASM to at least offer some protection (but that doesn't mean you can just run whatever, of course).

**Reply by 1661410107114145152525** [reply]
@bo0tzz Thanks a lot for the insights! ❤️
